vng_api_common/polymorphism.py

```diff
@@ -125,6 +127,11 @@ def __new__(cls, name, bases, attrs):
 class PolymorphicSerializer(serializers.HyperlinkedModelSerializer, metaclass=PolymorphicSerializerMetaclass):
     discriminator: Discriminator = None
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if args:
+            self.discriminator.instance = args[0]
```

Comment:
1. since this is a serializer, inheriting from DRF serializer, you should use `self.instance` instead of `args[0]`. `args[0]` is _really_ unclear, plus if it's passed via the kwargs `instance` you miss the argument as well.

2. There's actually no need to override the `__init__` to achieve what you're doing, it makes much more sense to add a `instance=None` kwarg to `Discriminator.to_internal_value` and pass it from `PolymorphicSerializer.to_internal_value`